@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -86,18 +88,46 @@ namespace SphericalProj {
     }
 
     /**
-     * Computes the far plane given the point where the upper right frustum edge intersects with
-     * the earth's sphere projected along the camera's view direction.
-     * @param worldSpaceIntersection - Furthest intersection point between frustum an sphere.
+     * Calculate distance to the nearest point where the near plane is tangent to the globe,
+     * projected onto the eye vector.
      * @param camera - The camera.
-     * @returns Camera pitch.
+     * @param halfVerticalFovAngle
+     * @param R - The globe radius.
+     * @returns The tangent point distance if the point is visible, otherwise `undefined`.
      */
-    export function getFarPlaneBasedOnFovIntersection(
-        worldSpaceIntersection: THREE.Vector3,
-        camera: THREE.PerspectiveCamera
-    ): number {
-        const toIntersection = tmpVectors[0].copy(worldSpaceIntersection).sub(camera.position);
-        return toIntersection.dot(camera.getWorldDirection(tmpVectors[1]));
+    export function getProjNearPlaneTanDistance(
+        camera: THREE.Camera,
+        halfVerticalFovAngle: number,
+        R: number
+    ): number | undefined {
+        const fwdDir = camera.getWorldDirection(tmpVectors[0]);
+        const nearPlaneTanPoint = tmpVectors[1].copy(fwdDir).multiplyScalar(-R);
+        const camToTan = nearPlaneTanPoint.sub(camera.position);
```

Comment:
`camToTanVec` isn't quite accurate, because a vector is a relative displacement whereas a point is a fixed location. But I won't block this if you want to keep it.